packages/hooks/src/useTrackedEffect/index.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef, DependencyList } from 'react';
+
+const useTrackedEffect = (effect, deps?: DependencyList) => {
+  const initDeps = useRef<DependencyList>();
+  const diffTwoDeps = (deps1, deps2) => {
+    //Let's do a reference equality check on 2 dependency list.
+    //If deps1 is defined, we iterate over deps1 and do comparison on each element with equivalent element from deps2
+    //As this func is used only in this hook, we assume 2 deps always have same length.
+    return deps1
+      ? deps1.map((_ele, idx) => (deps1[idx] != deps2[idx] ? idx : -1)).filter((ele) => ele >= 0)
+      : deps2
+      ? deps2.map((_ele, idx) => idx)
+      : [];
+  };
+  useEffect(() => {
+    let changes = diffTwoDeps(initDeps.current, deps);
+    initDeps.current = deps;
+    return effect(changes);
```

Comment:
It seems that `changes` only contains the indexes of changed elements in `deps`, which may not be enough in some circumstances. How about passing the previous `deps` and the current `deps`?

```ts
return effect(changes, initDeps.current, deps)
```

Reply:
I think it will make this hook over-complicated. Its main purpose should be indicating which dependencies has changed and where they are. Previous and current `deps `can be fetched easily from those returned indexes.

Reply:
I think it should be done，otherwise it is difficult for users to get previous deps.